src/serial.rs

```diff
@@ -993,9 +1076,9 @@ macro_rules! hal {
                     // the next statement, which starts the DMA transfer
                     atomic::compiler_fence(Ordering::Release);
 
-                    channel.start();
+                    self.channel.start();
 
-                    FrameReader::new(buffer, channel, usart.cr2.read().add().bits())
+                    FrameReader::new(buffer, self, usart.cr2.read().add().bits())
                 }
 
                 /// Checks to see if the USART peripheral has detected an idle line and clears
```

Comment:
Not sure what to do with the functions below, should they be attached to `FrameReader`?